server.js

```diff
@@ -237,12 +245,22 @@ function Server(config) {
         }
       },
       handler: function (req, reply) {
-        var sub = subscriptions.get(req.params.id)
-        if (sub) {
-          if (req.payload.pos) { sub.pos = req.payload.pos }
-          if (req.payload.notify_url) { sub.notify_url = req.payload.notify_url }
-        }
-        reply(sub || boom.notFound(req.params.id))
+        subscriptions.get(req.params.id)
+          .then(
+            function (sub) {
+              /* istanbul ignore else */
+              if (req.payload.pos) { sub.pos = req.payload.pos }
```

Comment:
> There is no "else"...is this some weirdness in instanbul that it tries to find on?

It checks both cases of all branches, which is pretty awesome actually, but in some cases like this, I don't care.

> Should we error out or ignore if they try to move pos backwards?

Maybe they're rewinding on purpose? I don't think we should guard against that.

I think the redis `LRANGE` [policy](http://redis.io/commands/lrange) is sensible. We might want to exclude negative values, but I like the rest.